src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,749 @@
+         *
+         * @return the Deadline as a String to be saved.
+         */
+        public String saveToData() {
+            if (this.done) {
+                return ("D | 1 | " + todo + " | " + doneBy.toString().replace("T", " "));
+            } else {
+                return ("D | 0 | " + todo + " | " + doneBy.toString().replace("T", " "));
+            }
+        }
+
+        @Override
+        public String toString() {
+            if (!this.done) {
+                return ("[D][ ] " + this.todo + " (by:" + dateFormat(this.doneBy) + ")");
+            } else {
+                return ("[D][X] " + this.todo + " (by:" + dateFormat(this.doneBy) + ")");
+            }
+        }
+    }
+
+    public static class Event extends Task {
+        /** The time at which the Event happens */
+        private LocalDateTime time;
+
+        /**
+         * Initiates a Event, time is set to 0000 if none is provided.
+         *
+         * @param s The task.
+         * @param time The time when the task occurs.
+         * @throws DateTimeParseException
+         */
+        public Event(String s, String time) throws DateTimeParseException {
+            super(s);
+            String[] split = time.split("\\s+");
+            if (split.length == 1) {
+                this.time = LocalDate.parse(time).atTime(0, 0);
+            } else if (split.length == 2) {
+                this.time = LocalDate.parse(split[0]).atTime(LocalTime.parse(split[1]));
+            } else {
+                throw new DateTimeParseException("", "", 1);
+            }
+        }
+
+        /**
+         * Returns a String form of the current Event to be saved onto the hard disk.
+         *
+         * @return the Event as a String to be saved.
+         */
+        public String saveToData() {
+            if (this.done) {
+                return ("E | 1 | " + todo + " | " + time.toString().replace("T", " "));
+            } else {
+                return ("E | 0 | " + todo + " | " + time.toString().replace("T", " "));
+            }
+        }
+
+        @Override
+        public String toString() {
+            if (!this.done) {
+                return ("[E][ ] " + this.todo + " (at:" + dateFormat(this.time) + ")");
+            } else {
+                return ("[E][X] " + this.todo + " (at:" + dateFormat(this.time) + ")");
+            }
+        }
+    }
+
+    public class Storage {
+
+        /** The path of the storage file where the TaskList is stored. */
+        private String filePath;
+
+        public Storage(String filePath) {
+            this.filePath = filePath;
+        }
+
+        /**
+         * Loads a TaskList from storage, if it exists.
+         * Else creates a new storage file.
+         *
+         * @return the TaskList stored on the hard disk.
+         */
+        public TaskList load() {
+            File taskData = new File(filePath);
+
+            TaskList leest = new TaskList();
```

Comment:
Haha, I get the joke! But maybe it would be better to name it to a more readible and intuitive variable name, like "taskList". In case some developers do not appreciate the humor. 😝